x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/hooks/use_fetch_agents_data.tsx

```diff
@@ -121,6 +128,9 @@ export function useFetchAgentsData() {
   ]);
 
   const [selectedTags, setSelectedTags] = useState<string[]>([]);
+  const [selectedOutputs, setSelectedOutputs] = useState<string[]>([]);
+  const [selectedPoliciesByOutputs, setSelectedPoliciesByOutputs] = useState<string[]>([]);
+  const [allOutputs, setOutputs] = useState<MinimalOutput[]>([]);
 
```

Comment:
Here I'm adding three different states:
- `allOutputs`: list of all the unique outputs present on the entire table. Needed to populate the new outputs dropdown filter
- `selectedOutputs`: sets the list of outputs ids selected by the above filter
- `selectedPoliciesByOutputs`: set the list of agent policies corresponding to the above output ids; these ids are retrieved by making an extra api call to the new endpoint `outputs/agent_policies` added in this PR, then these policies ids are passed to the getKuery to actually filter.